jsapp/js/components/activity/activityLogs.query.ts

```diff
@@ -1,116 +1,85 @@
-    log_type: log_type as AuditTypes,
-    metadata: metadata,
-  };
-};
-
-const curDate = new Date();
-const mockData: ActivityLogsItem[] = Array.from({length: 150}, (_, index) => {
-  curDate.setTime(curDate.getTime() - Math.random() * 1000000);
-  return {
-    id: index,
-    ...getRandomMockDescriptionData(),
-    date_created: moment(curDate).format('YYYY-MM-DD HH:mm:ss'),
-  };
-});
-// END OF MOCK GENERATION
-// =============================================================================
+import {fetchGet} from 'jsapp/js/api';
+import {endpoints} from 'jsapp/js/api.endpoints';
+import type {PaginatedQueryHookParams} from 'jsapp/js/universalTable/paginatedQueryUniversalTable.component';
 
 /**
  * Fetches the activity logs from the server.
  * @param limit Pagination parameter: number of items per page
  * @param offset Pagination parameter: offset of the page
  */
-const getActivityLogs = async (limit: number, offset: number) =>
-  new Promise<PaginatedResponse<ActivityLogsItem>>((resolve) => {
-    setTimeout(
-      () =>
-        resolve({
-          next: null,
-          previous: null,
-          count: mockData.length,
-          results: mockData.slice(offset, offset + limit),
-        } as PaginatedResponse<ActivityLogsItem>),
-      1000
-    );
+const getActivityLogs = async ({
+  projectId,
+  actionFilter,
+  limit,
+  offset,
+}: {
+  projectId: string;
+  actionFilter: string;
+  limit: number;
+  offset: number;
+}) => {
+  const params = new URLSearchParams({
+    limit: limit.toString(),
+    offset: offset.toString(),
   });
+  if (actionFilter) {
+    params.append('q', `action:${actionFilter}`);
+  }
+
+  const endpointUrl = endpoints.ASSET_HISTORY.replace(':asset_id', projectId);
+
+  return await fetchGet<PaginatedResponse<ActivityLogsItem>>(
+    `${endpointUrl}?${params}`,
+    {
+      errorMessageDisplay: t('There was an error getting one-time add-ons.'),
+    }
+  );
+};
 
 /**
  * Fetches the filter options for the activity logs.
+ *
+ * Filter options, for now, comes from ../api/v2/assets/[ assetId]]/history
+ * from the 'Filterable fields" section.
+ *
  */
 const getFilterOptions = async () =>
-  new Promise<KoboSelectOption[]>((resolve) => {
-    setTimeout(() => resolve(mockOptions), 1000);
+  [
+    'add-media',
+    'allow-anonymous-submissions',
+    'archive',
+    'connect-project',
+    'delete-media',
+    'delete-service',
+    'deploy',
+    'disable-sharing',
+    'disallow-anonymous-submissions',
+    'disconnect-project',
+    'enable-sharing',
+    'export',
+    'modify-imported-fields',
+    'modify-service',
+    'modify-sharing',
+    'modify-user-permissions',
+    'redeploy',
+    'register-service',
+    'replace-form',
+    'share-data-publicly',
+    'share-form-publicly',
+    'transfer',
+    'unarchive',
+    'unshare-data-publicly',
+    'unshare-form-publicly',
+    'update_content',
+    'update-name',
+    'update-settings',
+    'update-qa',
```

Comment:
This array of strings seems like should be a `const` somewhere. And I think it would be great to locate the BE file that also has these and add the path to that BE file in the FE comment

Reply:
Wait… Is this the same thing as the list from `jsapp/js/components/activity/activity.constants.ts`? It is! I think that list might require some updating (see other comment about finding BE file with the list of possible activities).

Reply:
For example `allow-anonymous-submissions` is missing (as table displays fallback message "user x did action y" through `ActivityMessage` component)

Reply:
At least in code I didn't find a place in BE with constants. Even in the [documentation page](https://github.com/kobotoolbox/kpi/blob/0cc5cd147415ffdf4ffd58c87a568f87bb97eced/kobo/apps/audit_log/views.py#L424) where it's listed it's a fixed string. I'll compare and add the missing ones, and in the future we gonna get these values from BE for the filter. Jackie is creating a task for that.